dvc/fs/fsspec_wrapper.py

```diff
@@ -11,6 +12,7 @@ class FSSpecWrapper(BaseFileSystem):
     def fs(self):
         raise NotImplementedError
 
+    @lru_cache(512)
```

Comment:
Why do we need it bt? Isinstance is too slow?

Reply:
It is mostly for `.isdir()/.isfile()` since they call `.exists()` + `.info()` which under the hood calls this function with the same arguments.